packages/ra-core/src/form/ValidationError.spec.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import { render, cleanup } from 'react-testing-library';
+
+import ValidationError from './ValidationError';
+
+const translate = jest.fn(key => key);
+
+jest.mock('../i18n', () => ({
+    useTranslate: () => translate,
+}));
+
+describe('ValidationError', () => {
+    afterEach(() => {
+        cleanup();
+        translate.mockClear();
+    });
+
+    it('It renders the error message translated if it is a string', () => {
+        const { getByText } = render(
+            <ValidationError error="ra.validation.required" />
+        );
+
+        expect(getByText('ra.validation.required')).toBeTruthy();
+        expect(translate).toHaveBeenCalledWith('ra.validation.required', {
```

Comment:
Bad testing strategy IMO. You'd better wrap your component with a TranslationProvider that translates the `ra.validation.required` message, and look for the translation in the render. Same for the other test.

Reply:
I don't get it. Can you explain why?

Reply:
Because your mock is testing internal method calls, instead of asserting that the result is what's expected from the end user's standpoint. In other terms, you're testing the implementation rather than the feature.